### PR TITLE
Feature/success timer game

### PR DESCRIPTION
**Pull request by kirilknysh**
Fix https://github.com/DaQuirm/css-quickdraw-redux/issues/222

image

### CODE REVIEW

File: app/facets/sandbox/sandbox.coffee [hunk 1]
```diff
@@ -39,12 +44,13 @@ class Sandbox
 
 		@log 'sandbox created'
 
-	cloneMatch: (match, player_id) ->
+	cloneMatch: (match, player_id, time_stamp) ->
```
Comment (DaQuirm) [inline app/facets/sandbox/sandbox.coffee]: Why pass `time_stamp` instead of generating it inside?

File: public/game/views/game-view.coffee [hunk 2]
```diff
@@ -56,9 +56,7 @@ GameView = (context) ->
 
 				nxt.Binding context.puzzle_solved, (isSolved) ->
 					if isSolved
```
Comment (DaQuirm) [inline public/game/views/game-view.coffee]: WDYT:
```coffeescript
CountdownCircleView
  if isSolved 
    context.solutionTimerViewModel
  else
    context.roundTimerViewModel
```

File: app/facets/sandbox/sandbox.coffee [hunk 3]
```diff
@@ -29,7 +29,12 @@ class Sandbox
 			]
 
 		@match = new nx.Cell
-			'<-': [@remote.match]
+			'<-': [
+				@remote.match
+				(match) ->
+					match.time_stamp = do Date.now
```
Comment (DaQuirm) [inline app/facets/sandbox/sandbox.coffee]: Arguably this should be done on Sandbox as it is more accurate (without the roundtrip etc.)

File: app/facets/sandbox/sandbox.coffee (hunk 1 above)
Comment (kirilknysh) [inline app/facets/sandbox/sandbox.coffee]: To have a pure `clone` method. It does nothing except cloning the object.

File: public/game/views/game-view.coffee (hunk 2 above)
Comment (kirilknysh) [inline public/game/views/game-view.coffee]: Will do

**Comment by DaQuirm** [comment]
Looks good!
One question: do we need `time_stamp` for all matches or just the correct ones?

**Comment by kirilknysh** [comment]
Well, for now only correct but who knows... :)

File: app/facets/sandbox/sandbox.coffee (hunk 1 above)
Comment (DaQuirm) [inline app/facets/sandbox/sandbox.coffee]: Ok, that's cool

**Pull request merged**